system/handlers/rules/expressions/question/UserHasResponse.cfc

```diff
@@ -0,0 +1,47 @@
+/**
+ *
+ * @expressionCategory formbuilder
+  * @expressionContexts user
+ * @feature            websiteusers
+ */
+component {
+
+	property name="rulesEngineOperatorService" inject="rulesEngineOperatorService";
+	property name="formBuilderService"         inject="formBuilderService";
+	property name="formBuilderFilterService"   inject="formBuilderFilterService";
+
+	/**
+	 * @question.fieldtype      formbuilderQuestion
+	 * @question.object         formbuilder_question
+	 * @question.item_type      ""
+	 *
+	 */
+	private boolean function evaluateExpression(
+		  required string question
+		,          boolean _has = true
+	) {
+		var filter = prepareFilters( argumentCollection = arguments	) ;
+
+		return formBuilderFilterService.evaluateQuestionUserHasResponse(
+			  argumentCollection = arguments
+			, userId             = payload.user.id
```

Comment:
I think we need to defend against this variable not existing. i.e. if there is no logged in user and this condition is run (in which case we can presumably immediately return:

```return !arguments._has;```

This would obviously need to be done for all the user based _expressions_ (the filters remain unchanged).